src/app/pages/AddProposal/AddProposal.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useState } from 'react'
+import { Link, useNavigate, useParams } from 'react-router-dom'
+import styled from 'styled-components'
+import Button from '../../components/Button/Button'
+import FormInput from '../../components/FormInput/FormInput'
+import DoubleChevronLeft from '../../Icons/DoubleChevronLeft'
+import type { Need, Proposal, Topic } from '../../types/types'
+
+const MAX_DESCRIPTION_LENGTH = 144
+
+type AddProposalProps = {
+  onSubmit: () => void
+  topics: Topic[]
+}
+
+export default function AddProposal({
+  onSubmit,
+  topics,
+}: AddProposalProps): JSX.Element {
+  const [description, setDescription] = useState('')
+  const descriptionDiff = MAX_DESCRIPTION_LENGTH - description.length
+
+  const params = useParams()
+  console.log(topics)
+
+  const topic = topics.find(topic => topic.id === params.topicId)
+  const needs = topic ? topic.needs : []
+
+  const nav = useNavigate()
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    onSubmit()
+  }
```

Comment:
Not sure if this is necessary. I think in line 48, you could also just apply `onSubmit={onSubmit}`.